doc/build-with-cmake.md

```diff
@@ -0,0 +1,8 @@
+Introduction
+============
```

Comment:
Nitpicking - I prefer the ATX-stype headers [1], as they make the header level explicit.

[ 1] http://pandoc.org/MANUAL.html#atx-style-headers